tests/integ/test_mxnet_train.py

```diff
@@ -72,6 +72,46 @@ def test_deploy_model(mxnet_training_job, sagemaker_session):
         predictor.predict(data)
 
 
+def test_deploy_model_with_update_endpoint(mxnet_training_job, sagemaker_session):
+    endpoint_name = 'test-mxnet-deploy-model-{}'.format(sagemaker_timestamp())
+
+    with timeout_and_delete_endpoint_by_name(endpoint_name, sagemaker_session):
+        desc = sagemaker_session.sagemaker_client.describe_training_job(TrainingJobName=mxnet_training_job)
+        model_data = desc['ModelArtifacts']['S3ModelArtifacts']
+        script_path = os.path.join(DATA_DIR, 'mxnet_mnist', 'mnist.py')
+        model = MXNetModel(model_data, 'SageMakerRole', entry_point=script_path,
+                           py_version=PYTHON_VERSION, sagemaker_session=sagemaker_session)
+        model.deploy(1, 'ml.t2.medium', endpoint_name=endpoint_name)
+        old_endpoint = sagemaker_session.describe_endpoint(EndpointName=endpoint_name)
+        old_config_name = old_endpoint['EndpointConfigName']
+
+        model.deploy(1, 'ml.m4.xlarge', update_endpoint=True, endpoint_name=endpoint_name)
+        new_endpoint = sagemaker_session.describe_endpoint(EndpointName=endpoint_name)['ProductionVariants']
+        new_production_variants = new_endpoint['ProductionVariants']
+        new_config_name = new_endpoint['EndpointConfigName']
+
+        assert old_config_name != new_config_name
+        assert new_production_variants['InstanceType'] == 'ml.m4.xlarge'
+        assert new_production_variants['InitialInstanceCount'] == 1
+        assert new_production_variants['AcceleratorType'] is None
```

Comment:
nit: use `assert not` instead.

Reply:
`not x` and `x is None` can result in different behavior depending on what `x` is. I think it's fine to be explicit here.

Reply:
Done.